pygmt/datatypes/dataset.py

```diff
@@ -153,6 +154,17 @@ def to_strings(self) -> np.ndarray[Any, np.dtype[np.str_]]:
             for segment in table.contents.segment[: table.contents.n_segments]:
                 if segment.contents.text:
                     textvector.extend(segment.contents.text[: segment.contents.n_rows])
+        if None in textvector:
+            # Workaround for upstream GMT bug reported in
+            # https://github.com/GenericMappingTools/pygmt/issues/3170.
+            msg = (
+                "The trailing text column contains `None' values and has been replace "
+                "with an empty string to avoid TypeError exceptions. "
+                "It's likely caused by an upstream GMT API bug. "
+                "Please consider reporting to us."
+            )
+            warnings.warn(msg, category=RuntimeWarning, stacklevel=1)
```

Comment:
```suggestion
            warnings.warn(msg, category=RuntimeWarning, stacklevel=2)
```

Reply:
`stacklevel=2` makes more sense here.

`stacklevel=1`:
```
pygmt/datatypes/dataset.py:166: RuntimeWarning: The trailing text column contains `None' values and has been replace with an empty string to avoid TypeError exceptions. It's likely caused by an upstream GMT API bug. Please consider reporting to us.
  warnings.warn(msg, category=RuntimeWarning, stacklevel=1)
```
`stacklevel=2`:
```
pygmt/clib/session.py:1940: RuntimeWarning: The trailing text column contains `None' values and has been replace with an empty string to avoid TypeError exceptions. It's likely caused by an upstream GMT API bug. Please consider reporting to us.
  return result.to_strings()
```